### PR TITLE
Allow byte-encoded userdata

### PR DESCRIPTION
**Pull request by vojtechcima**
CloudStack requires userdata to be byte-encoded. This patch allows to pass byte-encoded userdata.

### CODE REVIEW

File: cs/client.py [hunk 1]
```diff
@@ -54,6 +54,8 @@ def transform(params):
             continue
         if isinstance(value, string_type):
             continue
+        if key == "userdata" and isinstance(value, bytes):
```
Comment (brutasse) [inline cs/client.py]: Hi @vojtechcima!

Could you add `binary_type = str` in the `if PY2:` clause, `binary_type = bytes` in the `else` clause (at the top of the file) and check for `isinstance(value, binary_type)` instead? This ensures consistent behavior in Python 2 and 3.

Thank you!

**Comment by greut** [comment]
user-data are encoded in base64. I'm curious to see how you use it?

**Comment by vojtechcima** [comment]
Hi @greut, @brutasse,

I just wanted to be able to dynamically template and generate user-data within my Python3 scripts i.e. doing something like this:

```
USER_DATA = """
#cloud-config
runcmd:
    - apt-get update
"""

cs.deployVirtualMachine(
            # Other parameters here
            userdata=base64.b64encode(USER_DATA.encode("utf-8"))
)
```

raises value type error [here](https://github.com/exoscale/cs/blob/master/cs/client.py#L76).

**Comment by brutasse** [comment]
@vojtechcima ah ok… I guess you could do `cs.deployVirtualMachine(…, userdata=base64.b64encode(USER_DATA.encode("utf-8")).decode('utf-8'))` but that's a lot of encoding :)

I'm not sure about restricting this handling to `userdata`. Bytes are handled fine by the HTTP library and the signing process, are there other parameters that could be passed in bytes?

**Comment by brutasse** [comment]
An addition to `test_transform` with a bytes parameter would also be useful.

**Comment by greut** [comment]
> I'm not sure about restricting this handling to `userdata`.

Agreed.

File: tests.py [hunk 2]
```diff
@@ -187,16 +187,18 @@ def test_transform(self, get):
             'listvirtualmachinesresponse': {},
         }
         cs.listVirtualMachines(foo=["foo", "bar"],
-                               bar=[{'baz': 'blah', 'foo': 1000}])
+                               bar=[{'baz': 'blah', 'foo': 1000}],
+                               bytes_param='blah'.encode('utf-8'))
```
Comment (brutasse) [inline tests.py]: `b'blah'`

**Comment by brutasse** [comment]
Thanks @vojtechcima ! 2.1.5 is up on PyPI.

**Pull request merged**